src/key.js

```diff
@@ -106,17 +113,17 @@ class Key {
    * @returns {void}
    */
   clean () {
-    if (!this._buf || this._buf.length === 0) {
-      this._buf = Buffer.from(pathSepS)
+    if (!this._buf || this._buf.byteLength === 0) {
+      this._buf = pathSepB
     }
 
     if (this._buf[0] !== pathSep) {
-      this._buf = Buffer.concat([pathSepB, this._buf])
+      this._buf = Uint8Array.of(pathSep, ...this._buf)
     }
 
     // normalize does not remove trailing slashes
-    while (this._buf.length > 1 && this._buf[this._buf.length - 1] === pathSep) {
-      this._buf = this._buf.slice(0, -1)
+    while (this._buf.byteLength > 1 && this._buf[this._buf.byteLength - 1] === pathSep) {
+      this._buf = Uint8Array.of(...this._buf.slice(0, -1))
```

Comment:
🚨You end up allocating copying bytes twice here. I think what you want to do is `this._buf.subarray(0, -1)` that will create new `Uint8Array` view on the same underlying `ArrayBuffer`, which is what former code seemed to be doing.